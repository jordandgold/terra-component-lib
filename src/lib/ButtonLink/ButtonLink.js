import React from "react";
import PropTypes from "prop-types";
import "./ButtonLink.scss";

const ButtonLink = props => {
  const { link, className, text } = props;

  return (
    <React.Fragment>
      <a href={link} className={`ter-button ${className}`}>
        {text}
      </a>
    </React.Fragment>
  );
};

export default ButtonLink;

ButtonLink.propTypes = {
  link: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired
};
