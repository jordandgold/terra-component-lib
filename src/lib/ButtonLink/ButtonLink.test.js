import React from "react";

import { shallow } from "enzyme";

import ButtonLink from "./ButtonLink";

describe("ButtonLink", () => {
  let wrapper;
  let mockOnClick;
  let mockEventObject;

  beforeEach(() => {
    mockOnClick = jest.fn();
    wrapper = shallow(
      <ButtonLink className="test class" link="#" text="test text" />
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
