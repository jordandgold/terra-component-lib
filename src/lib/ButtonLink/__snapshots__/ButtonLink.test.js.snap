// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLink should match the snapshot 1`] = `
<Fragment>
  <a
    className="ter-button test class"
    href="#"
  >
    test text
  </a>
</Fragment>
`;
